scripts/functions/requirements/osx_brew

```diff
@@ -314,6 +314,11 @@ requirements_osx_brew_define_openssl()
   undesired_check openssl098
 
   case "$1" in
+    (ruby-3.2*|ruby-3.3*)
```

Comment:
This would fix a lot of macos issues, thanks!

However looking at https://www.rubyonmac.dev/openssl-versions-supported-by-ruby we should:
1. Add section below `(ruby-2.3*|...` with following content

```bash
(ruby-2*|ruby-3.0*)
        brew_openssl_package="openssl@1.1"
      # Needed for older openssl: https://bugs.ruby-lang.org/issues/18763
      export PKG_CONFIG_PATH="$(brew --prefix openssl@1.1)/lib/pkgconfig:$PKG_CONFIG_PATH"
```

And then add openssl@3 in the default `(*)` section.

Reply:
check it now

Reply:
lgtm